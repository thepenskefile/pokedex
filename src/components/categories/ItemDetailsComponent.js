// @flow

import React, { Fragment } from 'react';
// import { Link } from '@reach/router';
import { Image, Container, Box } from 'fannypack';

type Props = {
  data: Object
};

const ItemDetailsComponent = ({ data }: Props) => (
  <Container textAlign="left" marginLeft="10px" width="96%">
    <Fragment>
      <Box>Name: {data.name}</Box>
      <Box>Fling power: {data.fling_power}</Box>
      <Box>Effect: {data.effect_entries[0].effect}</Box>
      <Image src={data.sprites.default} />
    </Fragment>
  </Container>
);

export default ItemDetailsComponent;
