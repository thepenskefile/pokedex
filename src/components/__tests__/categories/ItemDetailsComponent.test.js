import React from 'react';
import ItemDetailsComponent from '../../categories/ItemDetailsComponent';
import renderer from 'react-test-renderer';

const defaultProps = {
  data: {
    name: 'mockName',
    fling_power: 'mockFlingPower',
    effect_entries: [{ effect: 'mockEffect' }],
    sprites: { default: 'mockImageUrl' },
    game_indices: [{ game_index: '1', generation: { name: 'mockGenerationName', url: 'mockUrl/1' } }],
    attributes: [{ name: 'mockAttributeName' }]
  }
};
it('renders correctly', () => {
  const component = renderer.create(<ItemDetailsComponent {...defaultProps} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
