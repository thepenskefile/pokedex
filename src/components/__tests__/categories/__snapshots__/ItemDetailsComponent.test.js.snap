// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="styled-sc-144wh3b-0 iUlbjp sc-bdVaJa eUADIL"
  style={
    Object {
      "marginLeft": "10px",
      "textAlign": "left",
      "width": "96%",
    }
  }
  width="96%"
>
  <div
    className="sc-ifAKCX jEBRGJ sc-bdVaJa eUADIL"
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <h1
      className="heading styled-cya4kb-0 iFEcqF sc-htoDjs hDVFrt sc-bdVaJa eUADIL"
    >
      MockName
    </h1>
    <img
      className="styled-sc-19nzt74-0 ivQgwC sc-eHgmQL ljDLPK sc-bdVaJa eUADIL"
      src="mockImageUrl"
    />
  </div>
  <div
    className="styled-sc-18btutf-0 eSjZSA sc-ifAKCX jEBRGJ sc-bdVaJa eUADIL"
    display="block"
    spacing="major-2"
    style={
      Object {
        "display": "block",
        "marginBottom": "20px",
      }
    }
  >
    <strong
      className="styled-sc-9qokh7-0 skDAJ sc-bdVaJa eUADIL"
      style={
        Object {
          "color": undefined,
        }
      }
    >
      Fling Power: 
    </strong>
    mockFlingPower
    <strong
      className="styled-sc-9qokh7-0 skDAJ sc-bdVaJa eUADIL"
      style={
        Object {
          "color": undefined,
        }
      }
    >
      Cost: 
    </strong>
  </div>
  <div
    className="styled-sc-18btutf-0 eSjZSA sc-ifAKCX jEBRGJ sc-bdVaJa eUADIL"
    spacing="minor-2"
    style={
      Object {
        "marginBottom": "20px",
      }
    }
  >
    <div
      className="styled__Tag-sc-1g01kou-2 iIJwdg sc-EHOje llGVFe sc-ifAKCX jEBRGJ sc-bdVaJa eUADIL"
      style={
        Object {
          "backgroundColor": "generation-",
        }
      }
    >
      <div
        className="styled__TagContent-sc-1g01kou-1 iuaGjz sc-bdVaJa eUADIL"
      >
        gen 
      </div>
    </div>
  </div>
  <span
    className="styled-sc-9qokh7-0 skDAJ sc-bdVaJa eUADIL"
    display="block"
    style={
      Object {
        "color": undefined,
        "display": "block",
      }
    }
  >
    mockEffect
  </span>
  <div
    className="sc-bdVaJa eUADIL"
    style={
      Object {
        "marginTop": "40px",
      }
    }
  >
    <h5
      className="heading styled-cya4kb-0 iFEcqF sc-htoDjs hDVFrt sc-bdVaJa eUADIL"
    >
      Attributes
    </h5>
    <span
      className="styled-sc-9qokh7-0 skDAJ sc-bdVaJa eUADIL"
      display="block"
      style={
        Object {
          "color": undefined,
          "display": "block",
        }
      }
    >
      <svg
        className="styled-huruki-0 gxxWxN sc-bdVaJa eUADIL"
        color="generation-"
        role="img"
        size={1}
        style={
          Object {
            "color": "generation-",
            "marginRight": "5px",
            "size": 1,
          }
        }
        viewBox="0 0 512 512"
      >
        <title>
          check
        </title>
        <path
          d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
      mockAttributeName
    </span>
  </div>
</div>
`;
